### PR TITLE
[ONNX] Added translation for string constants/inputs

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Added support for accepting string as inputs and constants

### Tickets:
 - 139685

### CODE REVIEW

File: src/frontends/onnx/frontend/src/core/tensor.cpp [hunk 1]
```diff
@@ -202,6 +202,25 @@ std::vector<char> Tensor::get_data() const {
...
+    if (m_tensor_proto->has_raw_data()) {
+        FRONT_END_THROW("Loading strings from raw data isn't supported");
+    }
+    if (m_tensor_proto->data_type() == TensorProto_DataType::TensorProto_DataType_STRING) {
+        return std::vector<std::string>(m_tensor_proto->string_data().begin(), m_tensor_proto->string_data().end());
+        std::vector<std::string> data(m_tensor_proto->string_data_size());
+        for (const auto& str : m_tensor_proto->string_data()) {
+            data.push_back(str);
+        }
+        return data;
```
[the remark this answers is not in the document]
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/core/tensor.cpp:219]: Good catch!

File: src/frontends/onnx/frontend/src/core/tensor.cpp [hunk 2]
```diff
@@ -202,6 +202,20 @@ std::vector<char> Tensor::get_data() const {
...
+template <>
+std::vector<std::string> Tensor::get_data() const {
+    if (has_external_data()) {
+        FRONT_END_THROW("External strings are not supported");
+    }
+    if (m_tensor_proto->has_raw_data()) {
+        FRONT_END_THROW("Loading strings from raw data isn't supported");
+    }
+    if (m_tensor_proto->data_type() == TensorProto_DataType::TensorProto_DataType_STRING) {
+        return std::vector<std::string>(m_tensor_proto->string_data().begin(), m_tensor_proto->string_data().end());
```
Comment (rkazants) [inline src/frontends/onnx/frontend/src/core/tensor.cpp:214]: can we use `detail::__get_data`? Is it already string objects in memory? can we create view tensor on it without recreation string objects?

Comment (gkrivor) [reply src/frontends/onnx/frontend/src/core/tensor.cpp:214]: I don't see an easy way to perform it without copying, and it isn't a purpose of this PR.
As I can see, when string is stored in a memory (disk or ram) - it isn't represented as a C-string. It looks like a bytes without an ending zero.
Sounds like you are talking about usage of std::string_view. If nothing changed - we haven't moved to C++17 standard yet. Moreover, I have some concerns string_view actually will solve the pointed issue, I don't remember I saw a code which allows to use string_view in constants in the core.
So, I think I'm obligated to use this approach to get strings from a protobuf format...

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 3]
```diff
@@ -6736,6 +6736,21 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_model_reduce_min_20_boolean) {
     test_case.run();
 }
 
+OPENVINO_TEST(${BACKEND_NAME}, onnx_string_input) {
+    const auto model = convert_model("string_input.onnx");
+    auto test_case = test::TestCase(model);
+    test_case.add_input<std::string>({"string1", "string2"});
+    test_case.add_expected_output<int64_t>({2});
+
```
Comment (rkazants) [inline src/frontends/onnx/tests/onnx_import.in.cpp:6744]: also recommend to add a test with Parameter (string format)->Result to check string output, also.

Comment (rkazants) [reply src/frontends/onnx/tests/onnx_import.in.cpp:6744]: may be it makes check Constant -> Result as well, if possible

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 4]
```diff
@@ -6736,6 +6736,25 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_model_reduce_min_20_boolean) {
     test_case.run();
 }
 
+OPENVINO_TEST(${BACKEND_NAME}, onnx_string_input) {
+    const auto model = convert_model("string_input.onnx");
+    auto test_case = test::TestCase(model);
+    test_case.add_input<std::string>({"strinpt1", "strinpt2"});
+    test_case.add_expected_output<int64_t>({2});
+    test_case.add_expected_output<std::string>({"strinpt1", "strinpt2"});
```
Comment (rkazants) [inline src/frontends/onnx/tests/onnx_import.in.cpp:6744]:
```suggestion
    test_case.add_input<std::string>({"string1", "string2"});
    test_case.add_expected_output<int64_t>({2});
    test_case.add_expected_output<std::string>({"string1", "string2"});
```

**Review by rkazants — APPROVED** [review]

File: src/frontends/onnx/tests/onnx_import.in.cpp (hunk 4 above)
Comment (gkrivor) [reply src/frontends/onnx/tests/onnx_import.in.cpp:6744]: It was an original variant and it is equal to string_constant test, I want to be sure we use correct model for testing.

**Pull request merged**